### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Geez14**

### CODE REVIEW

File: token.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_STRING,
+
+    // Keyworkd
+    TOKEN_KW_AUTO,
+
+    // PreProcessor
+    TOKEN_HASH_IF,
+
+    // increment-decrement
+    TOKEN_INCREMENT_PRE,
```
Comment (akash1047) [inline token.h]: How do you intend to check pre/post increment or decrement in lexical analysis?

File: token.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_INCREMENT_POST,
+    TOKEN_DECREMENT_PRE,
+    TOKEN_DECREMENT_POST,
+
+    // Arithmetic
+    TOKEN_ADD,
+    TOKEN_SUB,
+    TOKEN_DIV,
+    TOKEN_REM,
+    TOKEN_COMPLEMENT,
```
Comment (akash1047) [inline token.h]: TOKEN_TILDE

File: token.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_LEFT_SHIFT,
+
+    // Logical
+    TOKEN_LOGIC_NOT,
+    TOKEN_LOGIC_AND,
+    TOKEN_LOGIC_OR,
+
+    // Comparision
+    TOKEN_EQUAL,
+    TOKEN_NOTEQUAL,
```
Comment (akash1047) [inline token.h]: TOKEN_NOT_EQUAL

File: token.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_COMPLEMENT,
+    TOKEN_AND,
+    TOKKEN_OR,
+    TOKEN_XOR,
+    TOKEN_LEFT_SHIFT,
+    TOKEN_RIGHT_SHIFT,
+    TOKEN_LEFT_SHIFT,
+
+    // Logical
+    TOKEN_LOGIC_NOT,
```
Comment (akash1047) [inline token.h]: TOKEN_LOGICAL_NOT

File: token.h [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_LOGIC_AND,
```
Comment (akash1047) [inline token.h]: TOKEN_LOGICAL_AND

File: token.h [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_LOGIC_OR,
```
Comment (akash1047) [inline token.h]: TOKEN_LOGICAL_OR

File: token.h [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    TOKEN_SUB_ASSIGN,
+    TOKEN_STAR_ASSIGN,
+    TOKEN_DIV_ASSIGN,
+    TOKEN_REM_ASSIGN,
+    TOKEN_AND_ASSIGN,
+    TOKEN_OR_ASSIGN,
+    TOKEN_XOR_ASSIGN,
+    TOKEN_LEFT_SHIFT_ASSIGN,
+    TOKEN_RIGHT_SHIFT_ASSIGN,
+};
```
Comment (akash1047) [inline token.h:116]: TOKEN_TILDE_ASSIGN?

File: token.h [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    // DataType
+    TOKEN_IDENT,
+    TOKEN_INT,
+    TOKEN_FLOAT,
+    TOKEN_CHAR,
+    TOKEN_STRING,
+
+    // Keyworkd
+    TOKEN_KW_AUTO,
```
Comment (akash1047) [inline token.h:16]: where are the other keywords?

File: token.h [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+    // PreProcessor
+    TOKEN_HASH_IF,
```
Comment (akash1047) [inline token.h:52]: Where are the other preprocessors

**Review by akash1047 — CHANGES_REQUESTED** [review]

File: token.h (hunk 7 above)
Comment (Geez14) [reply token.h:116]: TILDE is unary operator, as the assignment operation is converted into -> "a op= b" -> "a = (type a) a op b"

Comment (akash1047) [reply token.h:116]: Right good catch. `~` is unary operator.

**Comment by akash1047** [comment]
Token variants are added.

**Pull request merged**